src/zimcheck/zimcheck.cpp

```diff
@@ -266,70 +266,73 @@ int zimcheck (const std::vector<const char*>& args)
         error.infoMsg("[INFO] Zimcheck version is " + std::string(VERSION));
 
         //Test 0: Low-level ZIM-file structure integrity checks
-        if(enabled_tests.isEnabled(TestType::INTEGRITY))
-            test_integrity(filename, error);
+        bool valid;
+        if(enabled_tests.isEnabled(TestType::INTEGRITY)) {
+            valid = test_integrity(filename, error);
+        }
+
 
-        // Does it make sense to do the other checks if the integrity
-        // check fails?
-        zim::Archive archive( filename );
-        error.addInfo("file_uuid",  stringify(archive.getUuid()));
+        if (valid) {
```

Comment:
You may get here with a garbage value of `valid` when the integrity check is excluded (e.g. when running a single check). After you fix it by initializing the variable to `true` it may make sense to rename it.

Also it may make sense to output an explicit message when the integrity check is skipped. Something like:

> WARNING: Integrity check is skipped. Any detected errors (in case zimcheck doesn't just crash) may in fact be due to corrupted/invalid data.

Reply:
Fixed (in the same commit).

I haven't keep the "(in case zimcheck doesn't just crash)" as it would also be a bug in libzim/zimcheck itself as we should not crash in case of invalid zim file.
Better have a issue opened in this case and not user thinking it is its zim file fault and go on.